thirdparty/kyaml/runfn/runfn.go

```diff
@@ -463,7 +397,11 @@ func (r *RunFns) ffp(spec runtimeutil.FunctionSpec, api *yaml.RNode, currentUser
 			"results-%v.yaml", r.resultsCount))
 		atomic.AddUint32(&r.resultsCount, 1)
 	}
-	if !r.DisableContainers && spec.Container.Image != "" {
+	fnConfig, err := r.getFunctionConfig(api)
```

Comment:
Can we call `api` something else that reflect its usage better ?

Also I found the signature for `getFunctionConfig` a bit awkward. My initial impression was that it updates the provided `api` to the read fn config or something but on reading the code found that it returns the `api` if there is no fnconfig provided.

Would suggest not to take any input for `getFunctionConfig`, instead here call `getFunctionConfig` only if r.fnConfig is specified otherwise assign the `api`.

```
       if r.fnConfig != "" {
                fnConfig, err  := r.getFunctionConfig()
       } else {
               fnConfig := api
       }

``` 

It will be simple and readable.

Reply:
This logic has been updated.